src/backend/api/meals.js

```diff
@@ -3,12 +3,68 @@ const router = express.Router();
 const knex = require('../database');
 
 router.get('/', async (req, res) => {
+  let meals = knex.select('meal.*').from('meal');
+
+  if ('maxPrice' in req.query) {
+    meals = meals.whereBetween('price', [0, req.query.maxPrice]);
+  }
+
+  if ('availableReservations' in req.query) {
+    meals = meals
+      .join('reservation', 'reservation.meal_id', '=', 'meal.id')
+      .groupBy('meal.id');
+    if (req.query.availableReservations === 'true') {
+      meals = meals.having(
+        'meal.max_reservations',
+        '>',
+        knex.raw('SUM(reservation.number_of_guests)')
+      );
+    } else {
+      meals = meals.having(
+        'meal.max_reservations',
+        '=',
```

Comment:
Just to be safe you could do

```suggestion
        '<=',
```

in case somehow we created too many reservations.